api/src/lib.rs

```diff
@@ -304,6 +310,82 @@ where
     Ok(StatusCode::Created)
 }
 
+async fn verify_verification_token<T>(
+    mut req: Request<State<T>>,
+) -> Result<StatusCode>
+where
+    T: Database,
+{
+    #[derive(Deserialize)]
+    struct VerifyVerifyBody
+    {
+        secret: String,
+        email: String,
+    }
+
+    let body: VerifyVerifyBody = req.body_json().await.map_err(server_err)?;
+
+    let db = &req.state().db;
+    let id = &req.ext::<VaultId>().unwrap().0;
+
+    verify_token::verify(db, id, &body.secret, VerifyMethod::Email(body.email))
+        .await
+        .map_err(server_err)?;
+
+    Ok(StatusCode::Ok)
+}
+
+async fn get_verification_status<T>(mut req: Request<State<T>>) -> Result
+where
+    T: Database,
+{
+    let db = &req.state().db;
+    let id = &req.ext::<VaultId>().unwrap().0;
+
+    let mut response = Response::builder(StatusCode::Ok)
+        .header("content-type", "application/json");
+
+    match verify_token::get(db, id).await.map_err(server_err) {
+        Ok(verify_token::PossibleToken::Verified) => {
+            Ok(response.body("true").build())
+        },
+        Ok(verify_token::PossibleToken::Unverified) => {
+            Ok(response.body("false").build())
+        },
+        Err(e) => Err(e),
+    }
+}
+
+async fn create_verification_token<T>(
+    mut req: Request<State<T>>,
+) -> Result<StatusCode>
+where
+    T: Database,
+{
+    #[derive(Deserialize)]
+    struct VerifyCreateBody
+    {
+        email: Option<String>,
+    }
+
+    let body: VerifyCreateBody = req.body_json().await.map_err(server_err)?;
```

Comment:
I doubt we can enforce unique email addresses easily, it would mean downloading and opening every file in the directory and checking.

Reply:
unless we key by email instead of pubkey. twilio also might do this for us. but yeah if we can't then I think it's okay leaving this as a future improvement for when we have a db